src/js/helpers/Repo.js

```diff
@@ -419,7 +420,35 @@ export default class Repo {
    * @return {Promise<void>}
    */
   async save(message) {
-    await this.addAll();
-    await this.commit(message, makeAuthor(this.user));
+    doingSave = true;
+    console.log("Repo.save(" + message + ")");
+    try {
+      // remove deleted files
+      const stagedFiles = await git.listFiles({
+        dir: this.dir
+      });
+
+      for (let i = 0, len = stagedFiles.length; i < len; i++) {
+        const filePath = path.join(this.dir, stagedFiles[i]);
+        const deleted = !fs.existsSync(filePath);
+        if (deleted) { // if deleted from file system, do double check if git needs to remove
+          const status = await this.status(stagedFiles[i]);
+          if (status === "*deleted") {
+            console.log("Repo.save() - removing deleted file: " + stagedFiles[i]);
+            await this.remove(stagedFiles[i]);
+          }
+        }
+      }
+
+      // add all modifications and new files
+      await this.add(".");
+      // commit staged files
+      await this.commit(message, makeAuthor(this.user));
+    } catch(e) {
+      doingSave = false;
+      console.error("Error on Repo.save(" + message + ")", e);
```

Comment:
This is redundant since we already log errors produced by `repo.save`. Let's remove this console log